chore: Change GEOSX to GEOS in a bunch of places

rrsettgast opened the pull request:
Changes were incrementally committed:
- [x] GEOSX_USE to GEOS_USE
- [x] GEOSX_ENABLE to GEOS_ENABLE
- [x] GEOSX_LA to GEOS_LA
- [x] GEOSX_TPL to GEOS_TPL
- [x] GEOSX_ to GEOS_
- [x] Update copyright statements

Sorry y'all....this PR is quite unreviewable. There should be no functional code changes.

rrsettgast committed:
GEOSX_USE to GEOS_USE

rrsettgast committed:
GEOSX_ENABLE to GEOS_ENABLE

rrsettgast committed:
GEOSX_LA to GEOS_LA

rrsettgast committed:
GEOSX_TPL to GEOS_TPL

rrsettgast committed:
GEOSX_ to GEOS_ excluding .py and .msub

rrsettgast committed:
change version number

rrsettgast committed:
GEOSX_TPL_DIR is env variable so keeping that for now

debugging ci script tar command

revert GEOS_TPL_DIR to GEOSX_TPL_DIR in ci_build_and_test_in_container.sh

revert GEOS_TPL_DIR to GEOSX_TPL_DIR in ci_build_and_test_in_container.sh

debugging

update LvArray

MelReyCG reviewed (approved):
What about the file names? Is it for a future PR?

CusiniM commented on examples/pygeosxExamples/hydraulicFractureWithMonitor/hydraulicFractureWithMonitor.py:
```diff
@@ -1,41 +1,41 @@
 from mpi4py import MPI
-import pygeosx
-from pygeosx_tools import wrapper
+import pygeos
```
I thoght there was an issue with an existing python package that has this same name. @cssherman ?

cssherman replied:
Yes, and it is a pre-requisite to a commonly used package that manages map projections (see https://pygeos.readthedocs.io/en/stable/ and https://pyproj4.github.io/pyproj/stable/).

untereiner replied:
Also `pySomething` or `something-py` are usually used for wrappers around code of other languages. If I understood correctly `pygeos` is not exactly a wrapper around geos. It is more an orchestrator. What we have is more a `rungeos` in python

rrsettgast replied:
ah...ok. well...I wasn't sure about changing the name of the python stuff. What are your suggestions @cssherman ?

cssherman replied:
Maybe `geos-wrapper` or `geos-python-driver`?

untereiner replied:
`geos-py-wrapper`  in case someday someone wants a `geos-lua-wrapper`... By the way, in this last case, it would require to add lua folder next to each python folder in the geos repository.

May I suggest to pull out all these python mixed C stuff into a new language mixed repository/package ? 
This new project should have `geos` as a external dependency.

rrsettgast replied:
I went with `geos_python_interface` from within python, and `geosPythonInterface` within `geos`. Now that I write that out, perhaps `geos_interface` from within python and `pythonInterface` within `geos` would be better....and a little less redundant.

rrsettgast replied:
I had linking problems when changing the python names...I reverted and will do that after we resolve the circular dependency.

CusiniM committed:
chore: update trilinos, hypre and fmt. (#3214)

Updated tpls:
- Trilinos: 15.1.1
- Hypre: hypre-v2.31.0-26-gf6cfb0355. Contains MGR improvements that required GEOS code changes.
- fmt 11.01.1

rrsettgast commented:
> What about the file names? Is it for a future PR?

We can add it to this PR....trying to see about one thing at a time.

rrsettgast committed:
workaround issue with env variable for GEOSX_TPL_DIR

Commit:
Update the  documentation with conventional commit prefixes (#3220)

jhuang2601 committed:
docs: update publications (#3216)

* Update Publications.rst

* Update Publications.rst

---------

Co-authored-by: Ryan Aronson <raronson0031@gmail.com>
Co-authored-by: Nicola Castelletto <38361926+castelletto1@users.noreply.github.com>
Co-authored-by: Randolph Settgast <settgast1@llnl.gov>

rrsettgast committed:
fix: Bugfix/semantic pull request fix (#3221)

* fix: do Not require successful conventional semantic job

---------

Co-authored-by: Lionel Untereiner <lionel.untereiner@geosiris.com>

Commit:
fix: another shot to fix workflow (#3225)

rrsettgast committed:
chore: Update blt, RAJA, CHAI, Caliper, adiak, conduit (#3218)

* update blt to 0.6.2

* update RAJA and CHAI to v2024.02.2

* update Caliper to 2.11.0

*update adiak to 0.4.0

* update conduit to 0.9.2

* change to new HDF import. use umpires fmt and add extra dependency to executable

* changes for change in raja interface

* move src/coreComponents/unitTests/dataRepository/testPacking.cpp to src/coreComponents/dataRepository/unitTests

* switched to syncronous packing/upacking to avoid test failures on CI image running on streak

cssherman commented:
@rrsettgast - it looks like you hit all of the required name changes for pygeos, including the manual linking step in the cmake.  The only other step I'd suggest is refactoring `pygeos` to `geos-python-driver`.

cssherman commented on src/docs/sphinx/advancedExamples/pygeosxExamples/inSituDataMonitor/AdvancedExample.rst:
```diff
@@ -20,34 +20,34 @@ This example requires two input xml files and one python script located at:
 Note that these paths will need to be updated if you run this problem outside of the example directory.
 
-.. literalinclude:: ../../../../../../examples/pygeosxExamples/hydraulicFractureWithMonitor/hydraulicFracture.xml
+.. literalinclude:: ../../../../../../examples/geosPythonInterface/hydraulicFractureWithMonitor/hydraulicFracture.xml
```
@rrsettgast - I think that these files are in is: `../../../../../../examples/geosPythonInterfaceExamples`

cssherman commented on scripts/setupPythonEnvironment.bash:
```diff
@@ -15,7 +15,7 @@ declare -a TARGET_PACKAGES=("geos-mesh-tools"
                             "geos-xml-tools"
                             "hdf5-wrapper"
-                            "pygeos-tools"
+                            "geos_python_interface-tools"
```
```suggestion
                            "geos-python-interface-tools"
```

cssherman commented on scripts/setupPythonEnvironment.bash:
```diff
@@ -223,7 +223,7 @@ then
     if [ ! -f "$BIN_DIR/geosx_preprocessed" ]
```
It may be worth changing this (and other instances) to geos_preprocessed

cssherman commented on src/geosPythonInterface/CMakeLists.txt:
```diff
@@ -0,0 +1,50 @@
+
+add_custom_command( TARGET geosPythonInterface
+                    PRE_BUILD
+                    COMMAND ${Python3_EXECUTABLE} -m venv ${CMAKE_BINARY_DIR}/lib/PYGEOS --system-site-packages )
```
We should also refactor `lib/PYGEOS` to something like `lib/GEOS_PYTHON` to avoid any potential import errors.

cssherman commented:
@rrsettgast - We'll also need to update some names in geosPythonPackages.  I've put together a PR here for that purpose:

https://github.com/GEOS-DEV/geosPythonPackages/pull/33

rrsettgast committed:
replace geosx with geos in varios variable names and comments

rrsettgast committed:
update copyright statements

rrsettgast committed:
add useful git command to copyrightPrepender.py

rrsettgast committed:
uncrustify

rrsettgast committed:
update PVTPackage

rrsettgast committed:
Merge branch 'develop' into cleanup/Macros-GEOSX_XXX-to-GEOS_XXX

rrsettgast committed:
update PVTPackage hash

rrsettgast committed:
touch main.cpp

rrsettgast committed:
update submodule

rrsettgast committed:
Merge branch 'cleanup/Macros-GEOSX_XXX-to-GEOS_XXX' of github.com:GEOS-DEV/GEOS into cleanup/Macros-GEOSX_XXX-to-GEOS_XXX
